pkgs/tools/package-management/nix/default.nix

```diff
@@ -112,6 +113,37 @@ let
     hash = "sha256-s1ybRFCjQaSGj7LKu0Z5g7UiHqdJGeD+iPoQL0vaiS0=";
   };
 
+  # Intentionally does not support overrideAttrs etc
+  # Use only for tests that are about the package relation to `pkgs` and/or NixOS.
+  addTestsShallowly = tests: pkg: pkg // {
+    tests = pkg.tests // tests;
+    # In case someone reads the wrong attribute
+    passthru.tests = pkg.tests // tests;
+  };
+
+  addFallbackPathsCheck = pkg: addTestsShallowly
+    { nix-fallback-paths =
+        runCommand "test-nix-fallback-paths-version-equals-nix-stable" {
+          paths = lib.concatStringsSep "\n" (builtins.attrValues (import ../../../../nixos/modules/installer/tools/nix-fallback-paths.nix));
+        } ''
+          if [[ "" != $(grep -v 'nix-${pkg.version}$' <<< "$paths") ]]; then
+            echo "nix-fallback-paths not up to date with nixVersions.stable (nix-${pkg.version})"
+            echo "The following paths are not up to date:"
+            grep -v 'nix-${pkg.version}$' <<< "$paths"
+            echo
+            echo "Fix it by running in nixpkgs:"
+            echo
+            echo "curl https://releases.nixos.org/nix/nix-${pkg.version}/fallback-paths.nix >nixos/modules/installer/tools/nix-fallback-paths.nix"
+            echo
+            exit 1
+          else
+            echo "nix-fallback-paths versions up to date"
+            touch $out
+          fi
+        '';
+    }
+    pkg;
```

Comment:
This is nixos-specific; it belongs in `nixos/` not in `nixpkgs/pkgs/`.

Reply:
It'll have to be added to the nix passthru tests in any case, just like we add nixosTests.

Reply:
> It'll have to be added to the nix attribute in any case, just like we add nixosTests.

That can be done in the `nixos` module for `nix`.

>  I can't think of a other place to put it.

I can: in `nixos/`.  These "fallback paths" are meaningless on non-NixOS uses of nixpkgs.

Reply:
The file is used by `nixos-rebuild` and `nix upgrade-nix` https://nixos.org/manual/nix/stable/command-ref/new-cli/nix3-upgrade-nix.html#description

Reply:
> The file is used by `nixos-rebuild`

Like I said, it's nixos-specific.



> The test has to be attached to nixVersions.stable in some way for ofborg to run it

ofborg runs plenty of tests from other parts of the tree, like `lib`, etc.